apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -159,9 +166,16 @@ export class OAuthService {
 			oauthResponse.jwt = jwtResponse;
 			return oauthResponse;
 		} catch (error) {
-			this.logger.log(error);
-			const system: System = await this.systemRepo.findById(systemId);
-			return this.getOAuthError(error as string, system.oauthConfig?.provider as string);
+			const oauthResponse: OAuthResponse = await this.systemRepo
+				.findById(systemId)
+				.then((system: System) => {
+					const provider = system.oauthConfig ? system.oauthConfig.provider : 'unknown-provider';
+					return this.getOAuthError(error as string, provider);
```

Comment:
I don't understand the point why we should avoid direct returns, but well then, let's do it. By the way, my IDE says it's code smell

Reply:
It come from the cleancode area. That say it is a self check for the developer and a documentation.

If the method is only a oneline, with a single function call, or simple logic it is fine for me to keep it as one line. For all other i want to follow the cleancode styles recommendations.

Reply:
Clean Code does not state anything about this Topic:
here is a list of general points and rules stated in 'Clean code' by Robert C. Martin:
https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29

Furthermore, there is actually a SonarQube rule inherited from PMD called [Unnecessary Local Before Return](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#UnnecessaryLocalBeforeReturn) that talks about this. It says:

> Avoid unnecessarily creating local variables.

This rule was [later replaced](http://jira.sonarsource.com/browse/SONARPMD-33) by SSLR rule [Variables should not be declared and then immediately returned or thrown](https://dev.eclipse.org/sonar/coding_rules#rule_key=squid%3AS1488%7Cs=createdAt%7Casc=false), which maintains the same position:

> Declaring a variable only to immediately return or throw it is a bad practice. Some developers argue that the practice improves code readability, because it enables them to explicitly name what is being returned. However, this variable is an internal implementation detail that is not exposed to the callers of the method. The method name should be sufficient for callers to know exactly what will be returned.

IntelliJ and Android Studio also has a warning for this situation:

> Variable used only in following return and can be inlined

This inspection reports local variables either used only in the very next return or exact copies of other variables. In both cases it's recommended to inline such a variable.

The real point from Clean Code here would be: 'Be consistent.', which is a valid point.

Nevertheless, we have already changed it, so this is resolved.